### DISCUSSION TITLE
Table Reflection and missing triggers

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#503] [Q&A] [answered]
I am attempting to use table reflection for a read only application that I am developing. I keep getting errors pertaining to `ValueError('Exception occurred while generating `logs` table: No matching trigger found')` however, this table an other don't have triggers. Is this something that you have encountered? Additionally, `piccolo` shouldn't really need to know about triggers on the tables, correct? Is there a way to not reflect those parts?

**Comment by dantownsend** [comment, marked as answer]
I haven't personally come across it yet, but the error is being raised here:

https://github.com/piccolo-orm/piccolo/blob/570b395d3adc81f344802484796c715aebdd39f3/piccolo/apps/schema/commands/generate.py#L173

The code is probably trying to work out the `ON UPDATE` and `ON DELETE` for a foreign key, and it's failing for some reason.

I think that bit of code needs changing, so it handles the exception gracefully, and instead outputs a warning.

Until a fix is released, you can patch your local version by changing that line of code, so it returns `None` instead of raising an exception.

**Reply by dantownsend** [reply]
Issue https://github.com/piccolo-orm/piccolo/issues/504

**Reply by theelderbeever** [reply]
Awesome! Thanks for the quick reply!
